webapp/src/features/intersections/map/map-legend.tsx

```diff
@@ -129,70 +99,155 @@ export const MapLegend = () => {
+          <Typography fontSize="small" sx={{ ml: 1 }}>
+            {key}
+          </Typography>
         </div>
       </React.Fragment>
     )
   }
 
   return (
-    <Accordion disableGutters defaultExpanded={false}>
-      <AccordionSummary>
-        <Typography variant="h5">Legend</Typography>
-      </AccordionSummary>
-      <AccordionDetails>
-        <Paper className="legend" style={{ userSelect: 'none' }}>
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">Signal Heads</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>{signalHeadIconsList}</div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Divider sx={{ borderRadius: 1 }} />
-
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">Lane Lines</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>{laneColorsList}</div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Divider sx={{ borderRadius: 1 }} />
-
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">Lane Connections</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>
-                {travelConnectionColorsList}
-              </div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">BSM Colors</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>{bsmColorsList}</div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Divider sx={{ borderRadius: 1 }} />
-        </Paper>
-      </AccordionDetails>
-    </Accordion>
+    <>
+      <Fab
+        id="map-legend-button"
+        style={{
+          position: 'absolute',
+          zIndex: 10,
+          top: theme.spacing(3),
+          right: theme.spacing(10),
+          backgroundColor: theme.palette.background.paper,
+        }}
+        size="small"
+        onClick={() => {
+          setOpen(!open)
+        }}
+      >
+        <VpnKeyOutlined />
+      </Fab>
+      <div
+        style={{
+          position: 'absolute',
+          zIndex: 10,
+          bottom: theme.spacing(3),
+          maxHeight: 'calc(100vh - 240px)',
+          right: 0,
+          width: open ? 600 : 50,
+          fontSize: '16px',
+          overflow: 'auto',
```

Comment:
The scrollbar should be for the contents, and leave the title at the top: 
<img width="302" alt="image" src="https://github.com/user-attachments/assets/268258e1-a824-4e33-91ad-f346f4cfcd83" />

Reply:
This applies to the other popup components as well

Reply:
Updated the popup components to only scroll content